Description: Fixed the build failure with Java 5
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: no
--- a/src/org/apache/xml/resolver/Catalog.java
+++ b/src/org/apache/xml/resolver/Catalog.java
@@ -1172,10 +1172,10 @@
    */
   public Iterator getPublicIDs() {
     Vector v = new Vector();
-    Enumeration enum = catalogEntries.elements();
+    Enumeration enumeration = catalogEntries.elements();
     
-    while (enum.hasMoreElements()) {
-	CatalogEntry e = (CatalogEntry) enum.nextElement();
+    while (enumeration.hasMoreElements()) {
+	CatalogEntry e = (CatalogEntry) enumeration.nextElement();
 	if (e.getEntryType() == PUBLIC) {
 	    v.add(e.getEntryArg(0));
 	}
